Fix audit issue logging by default peer address

pando85 opened the pull request:
By default log format include remote address that is taken from headers.
This is very easy to replace making log untrusted.

Changing default log format value `%a` to peer address we are getting
this trusted data always. Also, remote address option is maintianed and
relegated to `%{r}a` value.

Related  kanidm/kanidm#191.

pando85 committed:
Fix audit issue logging by default peer address

By default log format include remote address that is taken from headers.
This is very easy to replace making log untrusted.

Changing default log format value `%a` to peer address we are getting
this trusted data always. Also, remote address option is maintianed and
relegated to `%{r}a` value.

Related  kanidm/kanidm#191.

robjtede committed:
Merge branch 'master' into feature/log-source-ip

robjtede commented:
So if I have this right, you take issue with the fact that the current %a replacement uses the derived "Client IP" which could be taken from X-Forwarded-For headers which are obviously easy to replace if your proxy is misconfigured (or you don't have a proxy and the header is being sent manually).

There's a couple bits of terminology I want to nail down, document, and be strict to use in the right places.
- Client Address
- Peer Address
- Remote Address

Could you elaborate on the above so there is no confusion in reviewing? Might be worth adding doc comments to the ConnectionInfo fields to describe when and what they are populated with.

pando85 committed:
Rename peer/remote to remote_addr/realip_remote_addr

Change names to avoid naming confusions. I choose this accord to Nginx
variables and
[ngx_http_realip_module](https://nginx.org/en/docs/http/ngx_http_realip_module.html).

Add more specific documentation about security concerns of using Real IP
in logger.

pando85 commented:
@robjtede thanks for your comment!
I tried to be more clear about IP meanings choosing Nginx style names.

pando85 committed:
Merge branch 'master' into feature/log-source-ip

JohnTitor commented on src/middleware/logger.rs:
```diff
@@ -72,12 +72,21 @@ use crate::HttpResponse;
 ///
 /// `%{FOO}e`  os.environ['FOO']
 ///
+/// ### **\* security warning**
```
```suggestion
/// # Security
```
We usually use this header.

JohnTitor reviewed (commented):
I think this is a reasonable change, could you also update the changelog?

pando85 committed:
Rename security advertise header in doc

pando85 committed:
Add fix audit issue logging by default peer adress to changelog

pando85 committed:
Merge branch 'feature/log-source-ip' of github.com:pando85/actix-web into feature/log-source-ip

pando85 committed:
Merge branch 'master' into feature/log-source-ip

pando85 commented:
@JohnTitor both done.

Must I rebase all changes in one?

pando85 committed:
Merge branch 'master' into feature/log-source-ip

pando85 commented:
Any update?

JohnTitor reviewed (approved):
I missed this, sorry. Thanks!
